Lesson17/Gokhan/script.js

```diff
@@ -0,0 +1,210 @@
+// 12. Sum of Random Numbers: Write a do-while loop that generates random numbers 
+// between 1 and 5 and adds them together until the sum is greater than 20. 
+// Print the total number of iterations and the final sum.
+
+let sum = 0;
+let i = 1;
+
+do {
+    let randomNumber = Math.floor(Math.random() * 5) + 1;
+    sum += randomNumber;
+    i++;
+} while (sum <= 20);
+
+console.log(sum);
+console.log(i);
+
+// 13. Count Even Numbers: Write a for loop that generates and checks 100 random numbers between 1 and 50. Count how many even numbers are generated and print the count at the end.
+
+let count = 0;
+
+for (let i = 0; i < 100; i++) {
+    let randomNumber = Math.floor(Math.random() * 50) + 1;
+    if (randomNumber % 2 === 0) {
+        count++;
+    }
+}
+
+console.log(count);
+
+// 14. Find First Prime Number: Write a while loop that generates random numbers between 1 and 100 until it finds a prime number. 
+// Print the prime number 
+// (https://www.techtarget.com/whatis/definition/prime-number#:~:text=A%20prime%20number%20is%20a,%2C%2019%2C%2023%20and%2029.) 
+// and the number of iterations it took to find it.
+
+// ??? 
+
+
+// 15. Accumulate Multiples of Three: Write a while loop that continuously generates random numbers between 1 and 15. 
+// Stop the loop once the sum of numbers that are multiples of three exceeds 50. 
+// Print the final sum and the total number of multiples of three.
+
+let sum = 0;
+let totalMultiplesOfThree = 0;
+
+while (sum <= 50) {
+    let randomNumber = Math.floor(Math.random() * 15) + 1;
+    if (randomNumber % 3 === 0) {
+        sum += randomNumber;
+        totalMultiplesOfThree++;
+    }
+}
+
+console.log(sum);
+console.log(totalMultiplesOfThree);
+
+
+// 16. Array Element Doubling: Write a for loop that iterates over an array of numbers (e.g., [1, 2, 3, 4, 5]). 
+// For each iteration, double the value of each element and store the results in a new array. 
+// Print the original and the new array at the end.
+
+const numbers = [1, 2, 3, 4, 5];
+const doubledNumbers = [];
+
+for (let i = 0; i < numbers.length; i++) {
+    doubledNumbers.push(numbers[i] * 2);
+}
+
+console.log(numbers);
+console.log(doubledNumbers);
+
+
+// 17. Count Down Timer: Write a for loop that starts from 30 and counts down to zero, printing each number. 
+// However, it should print "Halfway there!" when it reaches 15, and "Almost done!" when it reaches 5.
+
+for (let i = 30; i >= 0; i--) {
+    if (i === 15) {
+        console.log("Halfway there!");
+    } else if (i === 5) {
+        console.log("Almost done!");
+    } else {
+        console.log(i);
+    }
+}
+
+
+// 18. Palindrome Checker: Write a for loop that checks whether a given string is a palindrome
+// (a word that reads the same backward as forward, like "radar" or "level").
+// The loop should iterate over the string and compare characters from the beginning and the end,
+// moving towards the center. Print whether or not the string is a palindrome.
+
+// ??? 
```

Comment:
```
let str = "level"; // String to check
str = str.toLowerCase(); // convert string to lower case

let isPalindrome = true; // Assume the string is a palindrome until proven otherwise
let left = 0; // Start index
let right = str.length - 1; // End index

for (; left < right; left++, right--) { // in this for statment the first `let i = 0;` (or similar) is omitted (`(; `) as we want to use variables defined outside the loop.
    if (str[left] !== str[right]) { // check if the characters at the same position both on the left and right sides are the same
        isPalindrome = false; // if not the same, the given string is not a palindrome 
        break; // Exit the loop if a mismatch is found
    }
}

if (isPalindrome) {
    console.log(`'${str}' is a palindrome.`);
} else {
    console.log(`'${str}' is not a palindrome.`);
}
```